DashAISoftware/DashAI
# Upload refactor

# Summary


Refactor the `Upload.jsx` component to use MUI. 
Two new functionalities were added: display the original name of the uploaded file and allow the user to delete the uploaded file in order to upload another one.

## Type of change



- Front end new feature.
- Refactoring.

## Changes


- Update `Upload.jsx` to use MUI components instead of the custom styles defined using `styled-components`
- When a file is loaded the user can see its original name.
- A button is added so the user can delete the loaded dataset and upload a new one.

## How to Test


1. Go to `DashAI/DashAI/front` and run `yarn json-server`
2. in another terminal in the same folder of step 1 run `yarn start`
3. Navigate to the datasets tab
4. Create a new dataset
5. Select "Tabular Classification" as your task
6. Select "Csv Data" as your dataloader
7. Upload a file by clicking on the button or using drag and drop.
8. You should see the original name of the file below the icon
9. Click on the "X" button in the top right of the drag and drop component
10. You should be able to upload a new dataset that replaces the previous one.

## Screenshots


[upload.webm](https://github.com/DashAISoftware/DashAI/assets/53657198/d337c8bf-b0ab-490d-a925-5d7681fb069d)


## Notes


In order to properly test the changes, new data and new endpoints were added to the json-server. These changes will be removed before merging with staging.
